src/rgw/rgw_user.cc

```diff
@@ -1876,7 +1876,14 @@ int RGWUser::add(const DoutPrefixProvider *dpp, RGWUserAdminOpState& op_state, o
 
   ret = check_op(op_state, &subprocess_msg);
   if (ret < 0) {
-    set_err_msg(err_msg, "unable to parse parameters, " + subprocess_msg);
+    if (is_populated() && (user_id.compare(op_state.get_user_id()) != 0)) {
+      set_err_msg(err_msg, "unable to create user " + user_id.to_str()
+		  + " because user id " + op_state.get_user_id().to_str()
+		  + " already exists with email "
+		  + op_state.get_user_email());
```

Comment:
by adding this condition here, we will end up printing this error message not only for duplicate email id, but also for other fields like duplicate `access_key` etc. 
The duplicate check already exists in `https://github.com/ceph/ceph/blob/master/src/rgw/rgw_user.cc#L1755` but it is masked by error returned from ` RGWUser::check_op() `

Reply:
Indeed. Perhaps it's better to have these checks in `RGWUser::check_op()` rather than in `RGWUser::execute_add()`.

Reply:
@smanjara ack, I've modified as such. Please take a look. Thanks.

Reply:
@smanjara ping